static/js/src/advantage/users/components/TableView/components.tsx

```diff
@@ -0,0 +1,192 @@
+import React from "react";
+import { format } from "date-fns";
+import CSS from "csstype";
+
+import Button from "@canonical/react-components/dist/components/Button";
+import Select from "@canonical/react-components/dist/components/Select";
+import { MainTableRow } from "@canonical/react-components/dist/components/MainTable/MainTable";
+
+import { User } from "../../types";
+import { UserRowVariant } from "./TableView";
+
+const DATE_FORMAT = "dd/MM/yyyy";
+
+type UserVariantProps = {
+  user: User;
+  variant: UserRowVariant;
+};
+
+type UserActionsProps = {
+  handleEditOpen: (id: string) => void;
+  handleEditSubmit: (id: string) => void;
+  handleCancel: () => void;
+} & UserVariantProps;
+
+const UserActions = ({
+  user,
+  variant,
+  handleEditOpen,
+  handleEditSubmit,
+  handleCancel,
+}: UserActionsProps) => {
+  return variant !== "editing" ? (
+    <Button
+      small
+      dense
+      onClick={() => handleEditOpen(user.id)}
+      disabled={variant === "disabled"}
+      aria-label={`Edit user ${user.email}`}
+    >
+      Edit
+    </Button>
+  ) : (
+    <>
+      <Button small dense onClick={handleCancel}>
+        Cancel
+      </Button>
+      <Button
+        small
+        dense
+        appearance="positive"
+        onClick={() => handleEditSubmit(user.id)}
+      >
+        Save
+      </Button>
+    </>
+  );
+};
+
+type UserEmailProps = UserVariantProps;
+
+const UserEmail = ({ user, variant }: UserEmailProps) => {
+  return (
+    <div style={{ display: "flex", alignItems: "center" }}>
+      {user.email}
+      {variant === "editing" ? (
+        <button
+          className="p-button--base u-no-margin--bottom"
+          style={{
+            marginLeft: "0.1rem",
+          }}
+        >
+          <i className="p-icon--delete" aria-label="delete"></i>
+        </button>
+      ) : null}
+    </div>
+  );
+};
+
+type UserRoleProps = UserVariantProps;
+
+const getVisibilityStyles = (isVisible: boolean): CSS.Properties => ({
```

Comment:
Typically we don't do any styles in JS but I'll leave this decision ultimately up to the project owners :)

Reply:
CSS-in-JS fan here 🙈 
But I do try to reuse the existing styles/use vanilla classes whenever possible.